client.go

```diff
@@ -18,8 +18,17 @@ func (p *Provider) NewSession() error {
 	if p.MaxRetries == 0 {
 		p.MaxRetries = 5
 	}
-	sess, err := session.NewSession(&aws.Config{
-		MaxRetries: aws.Int(p.MaxRetries),
+	sess, err := session.NewSessionWithOptions(session.Options{
+		// Specify profile to load for the session's config
+		Profile: p.AwsProfile,
+
+		// Provide SDK Config options, such as Region.
+		Config: aws.Config{
+			MaxRetries: aws.Int(p.MaxRetries),
+		},
+
+		// Force enable Shared Config support
```

Comment:
Why?

Reply:
perhaps @danlsgiga can help answer.

Reply:
@mholt, I believe this is required to enable multiple profiles to use the same `~/.aws/config` file among all specified profiles... otherwise, it requires a specific block in the config file for each profile.

Reply:
Gotcha. Ok, either the comment should explain that or we can remove it if it's obvious to people familiar with the library/service.

Reply:
Kewl... @PhoenixPeca can you please remove the comment? I think this is redundant and I agree with Matt here!

Once that is done, it LGTM and you can merge it @mholt! ;)

Reply:
Hi @danlsgiga , sorry for late reply. I will edit now, but i havent had the time to test this with your recent question about EC2 instance profiles. Will do my best to test as soon as possible.

Reply:
> Have you tested this using EC2 Instance profiles to see how it behaves (not providing any profile)?

Hi @danlsgiga , so far working fine on my side. :D